lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -1060,16 +1068,16 @@ private StringBuilder getGeneralLockStats() {
     @Override
     public void close() {
         if (isShutDown.compareAndSet(false, true)) {
-            executor.shutdownNow();
-            wakeIndefiniteBlockers();
-            callOnClose.run();
+            shutDownTask.run();
         }
     }
 
-    private void wakeIndefiniteBlockers() {
-        for (Thread blocked : indefinitelyBlockingThreads) {
-            blocked.interrupt();
+    private void shutDown(InjectableResource<ExecutorService> executor, Runnable callOnClose) {
+        if (!executor.isInjected()) {
```

Comment:
with the `Owned` suggestion, it'll look like:

```java
private void shutDown() {
    if (executor.isOwned()) {
        executor.resource().shutdownNow();
    }
    indefinitelyBlockingThreads.forEach(Thread::interrupt);
	callOnClose.run();
}
```

with the `executor` and `callOnClose` remaining as fields

Reply:
or if wrapped:

```java
private void shutDown() {
  executor.shutdownNow();
  indefinitelyBlockingThreads.forEach(Thread::interrupt);
  callOnClose.run();
}
```